### PR TITLE
html: Add support for autoclosing of tags

### LINKED ISSUE #5267: JSX/HTML tag auto-close feature
### Check for existing issues

- [X] Completed

### Is your feature request related to a problem?

It's a bit frustrating to add the close tag at the end of the `jsx` element or `html` element. So, having that auto-close tag functionality will definitely help there.

### Describe the solution you'd like

Something like the `vscode`  setting for auto-close tag feature. As soon as you type the opening of the tag, the end of the tag comes as well.

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by osiewicz**
Fixes #5267 
TODO:
- [x] Publish our fork of vscode-langservers-extracted on GH and wire that through as a language server of choice for HTML extension.
- [x] Figure out how to prevent edits made by remote participants from moving the cursor of a host.

Release Notes:

- Added support for autoclosing of HTML tags in local projects.

### CODE REVIEW

File: extensions/html/src/html.rs [hunk 1]
```diff
@@ -2,7 +2,7 @@ use std::{env, fs};
 use zed_extension_api::{self as zed, Result};
 
 const SERVER_PATH: &str =
-    "node_modules/vscode-langservers-extracted/bin/vscode-html-language-server";
+    "/Users/hiro/Projects/vscode-langservers-extracted/bin/vscode-html-language-server";
```
Comment (zetashift) [inline extensions/html/src/html.rs:5]: Hardcoded path here still? (Sorry if you were already aware of it.)

Comment (osiewicz) [reply extensions/html/src/html.rs:5]: No worries; yeah, this PR is still WIP

**Comment by BZBaXraM** [comment]
Add it!

**Comment by osiewicz** [comment]
For the record, this is not gonna work for remote participants of a pairing session, though we should be able to make it so with a new RPC call.

**Comment by osiewicz** [comment]
Relevant repositories:
- https://github.com/zed-industries/vscode-langservers-extracted - bundling of HTML extension
- https://github.com/zed-industries/vscode-languageserver-node - I needed to wire through document change tracking in VSC document manager.
- https://github.com/zed-industries/vscode - the change itself - I bumped the protocol to include changes from version 3.18 and I added a listener for document changes that sends out a snippet workspace edit if tag autoclosing should take place.

All of this should work outside of Zed. Note that I'm not a npm guru (far from it), so the way deps are wired through to make installing work via github might be a bit of an eyesore.

**Comment by ThatOneCalculator** [comment]
Does this also apply to Vue files?

**Comment by mariansimecek** [comment]
Hi, it seems there is a problem with autocompleting tags when it shouldn't. This is a TypeScript file:

Screenshot from 2024-05-21 10-14-19

**Comment by osiewicz** [comment]
Thanks for reporting this @mariansimecek , I'll follow up shortly with a fix.

**Comment by mariansimecek** [comment]
@osiewicz I found another problem, now inside `tsx` file.  It autocompletes tag when writing lambda function. 
Screenshot from 2024-05-21 10-50-48
Screenshot from 2024-05-21 10-50-52

**Comment by osiewicz** [comment]
Yeah, I think we should just roll back completions in TS/JS files for the time being. Bummer.

**Comment by nryberg** [comment]
While I love the autoclosing tags, is there a way I can turn this off on a per file basis?  I have partials that intentionally leave dangling tags that are closed by another partial. Kind of a "please stop helping" moment. 

Have:
```html
    <main id="home" class="documentation"></main>
  </body>
</html>
```

Want:
```html
    <main id="home" class="documentation">
```

**Pull request merged**